src/expand.rs

```diff
@@ -0,0 +1,175 @@
+use crate::parse::{parse_decimal, parse_id};
+use crate::Captures;
+use std::borrow::Cow;
+use std::io;
+use std::mem;
+
+/// A set of options for expanding a template string using the contents
+/// of capture groups.  Create using the `builder` method.
```

Comment:
I didn't mind having the builder btw, you can add it back if you want. Otherwise, remove this from the docs :).

Reply:
I decided the builder API did exposes too many implementation details without adding useful generality. In particular, I think Perl and Sed syntax might be good additions at some point, and I don't thing either could be reproduced faithfully with the builder API. The default syntax is perfectly fine for any use case that doesn't require compatibility with another implementation, and adding an API that _almost_ allows emulating another implementation seems counterproductive.